sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CodeGenerationSuite.scala

```diff
@@ -97,6 +97,38 @@ class CodeGenerationSuite extends SparkFunSuite with ExpressionEvalHelper {
     assert(actual(0) == cases)
   }
 
+  test("SPARK-18091: split large if expressions into blocks due to JVM code size limit") {
+    val row = create_row("afafFAFFsqcategory2dadDADcategory8sasasadscategory24", 0)
```

Comment:
I think I didn't make this clear. So the fix I've made in If expression is actually required in all of the expressions so that all the expressions break their generated code into methods whenever it grows beyond an appropriate threshold. Since the fix in this pull request is only in If expression, a working testcase with the fix will be one which consists (recursively) of If expressions. Because otherwise the test will need to be tuned so that the size of If expression's code is greater than threshold but of it's predicate, trueValue and falseValue doesn't grow beyond threshold